Bottom: bcf7f681b80f3e20d2bc02e543338ea777582e12
Top:    2472e1815dbc1c7df622e8d3ef39877336879a25
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 16:19:39 +0000

Refresh of draw-pin-pad-names-first-so-th

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index af3c01a..b266b04 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1065,7 +1065,21 @@ draw_pin (PinType *pin, bool draw_hole)
 static int
 pin_callback (const BoxType * b, void *cl)
 {
-  draw_pin ((PinType *)b, TEST_FLAG (THINDRAWFLAG, PCB));
+  PinType *pin = (PinType *) b;
+
+  if (!TEST_FLAG (HOLEFLAG, pin) && TEST_FLAG (DISPLAYNAMEFLAG, pin))
+    _draw_pv_name (pin);
+  draw_pin (pin, TEST_FLAG (THINDRAWFLAG, PCB));
+  return 1;
+}
+
+static int
+pin_name_callback (const BoxType * b, void *cl)
+{
+  PinType *pin = (PinType *) b;
+
+  if (!TEST_FLAG (HOLEFLAG, pin) && TEST_FLAG (DISPLAYNAMEFLAG, pin))
+    _draw_pv_name (pin);
   return 1;
 }
 
@@ -1175,8 +1189,11 @@ pad_callback (const BoxType * b, void *cl)
   PadType *pad = (PadType *) b;
   int *side = cl;
 
-  if (ON_SIDE (pad, *side))
+  if (ON_SIDE (pad, *side)) {
+    if (TEST_FLAG (DISPLAYNAMEFLAG, pad))
+      draw_pad_name (pad);
     draw_pad (pad);
+  }
   return 1;
 }
 
@@ -1427,6 +1444,9 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
 
       /* Draw pins, vias and pads on this layer */
       if (!global_view_2d && rv) {
+        if (PCB->PinOn &&
+            (group == bottom_group || group == top_group))
+          r_search (PCB->Data->pin_tree, screen, NULL, pin_name_callback, Layer);
         if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, pin_inlayer_callback, Layer);
         if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, via_inlayer_callback, Layer);
         if (PCB->PinOn && group == top_group)
